Backend/auction-app/src/main/java/com/example/auctionapp/service/UserService.java

```diff
@@ -0,0 +1,149 @@
+package com.example.auctionapp.service;
+
+import com.example.auctionapp.dto.UserDto;
+import com.example.auctionapp.exception.BadRequestException;
+import com.example.auctionapp.exception.NotFoundException;
+import com.example.auctionapp.model.Role;
+import com.example.auctionapp.model.User;
+import com.example.auctionapp.repository.BaseRepository;
+import com.example.auctionapp.repository.UserRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@Transactional
+public class UserService implements IBaseService<UserDto> {
+
+    private BaseRepository<User> baseUserRepository;
+    private UserRepository userRepository;
```

Comment:
By making `UserRepository` extend `BaseRepository<User>`, we would have all the functionalities of these classes user-wise in one class. This is why we created `BaseRepository<T>` in the first place. Specific repositories will have their own logic, but also have logic common to all repositories, which is inherited from `BaseRepository`